### PR TITLE
Assign Gear attributes for adding magazines to weapons

### PR DESCRIPTION
**Pull request by Boberro**
(new PR) As Arma works right now, if we want to add magazine to an unit with addMagazine command, unit has to have free space in an inventory for said magazine, even if it would go into the weapon. Eg. if unit doesn't have a backpack, we can't add a missile to it as it wont fit in unit's vest.
I've added primaryMagazine[], secondaryMagazine[] and sidearmMagazine[] attributes (only one item in array will work in most cases, but it's an array in case of grenade launchers) to loadout classes, so one does not need to use attachment[] attribute (if attachment is used, bipod/silencer etc are ignored).
Supported by autotest.

### CODE REVIEW

File: addons/assigngear/functions/fn_replacePrimaryWeapon.sqf [hunk 1]
```diff
@@ -17,7 +17,7 @@
  * Replaces a units' primary weapon
```
Comment (Freddo3000) [inline addons/assigngear/functions/fn_replacePrimaryWeapon.sqf]: Function header needs to be updated

File: addons/assigngear/functions/fn_replaceSecondaryWeapon.sqf [hunk 2]
```diff
@@ -14,7 +14,7 @@
  * Replaces a units' secondary weapon
```
Comment (Freddo3000) [inline addons/assigngear/functions/fn_replaceSecondaryWeapon.sqf]: Function header needs to be updated

File: addons/assigngear/functions/fn_replaceSecondaryWeapon.sqf [hunk 3]
```diff
@@ -24,6 +24,13 @@ private _weapon = toLower selectRandom _weaponArray;
 // Exit if weapon was set to "default"
 if (_weapon isEqualTo "default") exitWith {};
 
+if(isNil "_magazineArray") then {_magazineArray = []};
+{
+    if (!isNil "_x" && _x != "" ) then {
```
Comment (Freddo3000) [inline addons/assigngear/functions/fn_replaceSecondaryWeapon.sqf]: `!isNil "_x"` I don't believe this is necessary

File: addons/assigngear/functions/fn_replaceSidearmWeapon.sqf [hunk 4]
```diff
@@ -14,7 +14,7 @@
  * Replaces a units' sidearm weapon
```
Comment (Freddo3000) [inline addons/assigngear/functions/fn_replaceSidearmWeapon.sqf]: Function header needs to be updated

File: addons/assigngear/functions/fn_replaceSidearmWeapon.sqf [hunk 5]
```diff
@@ -24,6 +24,13 @@ private _weapon = selectRandom _weaponArray;
 // Exit if weapon was set to "default"
 if (_weapon isEqualTo "default") exitWith {};
 
+if(isNil "_magazineArray") then {_magazineArray = []};
+{
+    if (!isNil "_x" && _x != "" ) then {
```
Comment (Freddo3000) [inline addons/assigngear/functions/fn_replaceSidearmWeapon.sqf]: `!isNil "_x"` I don't believe this is necessary

File: addons/assigngear/functions/fn_replacePrimaryWeapon.sqf [hunk 6]
```diff
@@ -46,7 +46,14 @@ if(count _this > 3) then {
         if (!isNil "_x" && _x != "" ) then {
             _unit addPrimaryWeaponItem _x;
         };
-    } forEach [_scope,_bipod,_attachment,_silencer];;
+    } forEach [_scope,_bipod,_attachment,_silencer,_magazine];
+
+    if(isNil "_magazineArray") then {_magazineArray = []};
+    {
+        if (!isNil "_x" && _x != "" ) then {
```
Comment (Freddo3000) [inline addons/assigngear/functions/fn_replacePrimaryWeapon.sqf]: `!isNil "_x"` I don't believe this is necessary

File: addons/assigngear/functions/fn_replacePrimaryWeapon.sqf [hunk 7]
```diff
@@ -46,7 +46,14 @@ if(count _this > 3) then {
         if (!isNil "_x" && _x != "" ) then {
             _unit addPrimaryWeaponItem _x;
         };
-    } forEach [_scope,_bipod,_attachment,_silencer];;
+    } forEach [_scope,_bipod,_attachment,_silencer,_magazine];
+
+    if(isNil "_magazineArray") then {_magazineArray = []};
```
Comment (Freddo3000) [inline addons/assigngear/functions/fn_replacePrimaryWeapon.sqf]: Probably better to edit params to be
` params ["_unit","_weaponArray","_scopeArray","_bipodArray","_attachmentArray","_silencerArray", ["_magazineArray",[]]];`

Infact, that goes for all of the parameters there

File: addons/assigngear/functions/fn_replaceSecondaryWeapon.sqf [hunk 8]
```diff
@@ -14,7 +14,7 @@
  * Replaces a units' secondary weapon
  */
  #include "\x\tmf\addons\assignGear\script_component.hpp"
- params ["_unit","_weaponArray","_items"];
+ params ["_unit","_weaponArray","_items","_magazineArray"];
```
Comment (Freddo3000) [inline addons/assigngear/functions/fn_replaceSecondaryWeapon.sqf]: Probably better to edit params to be
` params ["_unit","_weaponArray","_items",["_magazineArray",[]]];`

Infact, that goes for all of the parameters there

File: addons/assigngear/functions/fn_replaceSidearmWeapon.sqf [hunk 9]
```diff
@@ -14,7 +14,7 @@
  * Replaces a units' sidearm weapon
  */
  #include "\x\tmf\addons\assignGear\script_component.hpp"
- params ["_unit","_weaponArray","_items"];
+ params ["_unit","_weaponArray","_items","_magazineArray"];
```
Comment (Freddo3000) [inline addons/assigngear/functions/fn_replaceSidearmWeapon.sqf]: Probably better to edit params to be
`params ["_unit","_weaponArray","_items",["_magazineArray",[]]];`

Infact, that goes for all of the parameters there

**Comment by Boberro** [comment]
Caching update completely fucked this up.

**Pull request closed without merging**